OpenRA.Game/Traits/ActivityUtils.cs

```diff
@@ -20,29 +20,21 @@ public static class ActivityUtils
 	{
 		public static Activity RunActivity(Actor self, Activity act)
 		{
-			// PERF: If there are no activities we can bail straight away and save ourselves a call to
-			// Stopwatch.GetTimestamp.
+			// PERF: If there are no activities we can bail straight away and save ourselves a potential call to
+			// Stopwatch.GetTimestamp in perfLogging.Start().
 			if (act == null)
 				return act;
 
 			// PERF: This is a hot path and must run with minimal added overhead.
-			// Calling Stopwatch.GetTimestamp is a bit expensive, so we enumerate manually to allow us to call it only
-			// once per iteration in the normal case.
-			// See also: DoTimed
-			var longTickThresholdInStopwatchTicks = PerfTimer.LongTickThresholdInStopwatchTicks;
-			var start = Stopwatch.GetTimestamp();
+			// Calling Stopwatch.GetTimestamp is a bit expensive, so we enumerate manually to allow us to call
+			// perfLogging.Log only once per iteration in the normal case.
+			var perfLogging = new SimulationPerfLogging();
```

Comment:
Would there be any value in making `self.World.TraitDict.perfLogging` public and then reusing that here?

Reply:
No idea.

I went with moving it to its own file, so I guess it doesn't matter anymore (unless we change it back).